src/RigidBody/Joints.cpp

```diff
@@ -409,6 +409,80 @@ int rigidbody::Joints::GetBodyRbdlIdToBiorbdId(
 }
 
 
+std::vector<std::vector<unsigned int> > rigidbody::Joints::GetChildRbdlIdx() const
+{
+    return  this->mu;
+}
+
+std::vector<std::vector<unsigned int> > rigidbody::Joints::GetSubTrees()
+{
+    // initialize subTrees
+    std::vector<std::vector<unsigned int> > subTrees;
+    std::vector<unsigned int> subTree_empty;
+    for (unsigned int j=0; j<this->mu.size(); ++j) {
+        subTrees.push_back(subTree_empty);
+    }
+
+    // Get all dof without parent
+    std::vector<unsigned int> dof_with_no_parent_id;
+    for (unsigned int i=1; i<this->mu.size(); ++i) { // begin at 1 because 0 is its own parent in rbdl.
+      if (this->lambda[i]==0) {
+          dof_with_no_parent_id.push_back(i);
+      }
+    }
+
+    // Get all subtrees of dofs without parents
+    for (unsigned int i=0; i<dof_with_no_parent_id.size(); ++i) {
+      unsigned int dof_id = dof_with_no_parent_id[i];
+
+      // initialize subTrees_temp
+      std::vector<std::vector<unsigned int> > subTrees_temp;
+      for (unsigned int j=0; j<this->mu.size(); ++j) {
+          subTrees_temp.push_back(subTree_empty);
+      }
+
+      std::vector<std::vector<unsigned int> > subTrees_temp_filled = recursiveSubTrees(subTrees_temp, dof_id);
+      for (unsigned int j=0; j<subTrees_temp.size(); ++j) {
+          if (!(subTrees_temp_filled[j].empty())) {
+                subTrees[j].insert(subTrees[j].end(),
+                                   subTrees_temp_filled[j].begin(),
+                                   subTrees_temp_filled[j].end());
+            }
+      }
+    }
+    subTrees.erase(subTrees.begin());
+
+    return  subTrees;
+}
+
+std::vector<std::vector<unsigned int> > rigidbody::Joints::recursiveSubTrees(
+        std::vector<std::vector<unsigned int> >subTrees,
+        unsigned int idx)
```

Comment:
If feel that recursiveSubTrees should not be called by a lambda user, but only by the class itself. This one should definitely be a protected method

Reply:
done.